project/Build.scala

```diff
@@ -1346,21 +1347,24 @@ object Build {
 
         val dottyLibRoot = projectRoot.relativize(dottyManagesSources.toPath.normalize())
 
+        def generateDocTask =
+          generateDocumentation(
+            roots, "Scala 3", dest.getAbsolutePath, "master",
+            Seq(
+              "-comment-syntax", "wiki",
+              s"-source-links:docs=github://lampepfl/dotty/master#docs",
+              "-doc-root-content", docRootFile.toString,
+              "-Ydocument-synthetic-types"
+            ) ++ (if (justAPI) Nil else Seq("-siteroot", "docs", "-Ylegacy-api-layout")))
```

Comment:
Why do we want to use `-Ylegacy-api-layout` here? Why not use the new way?

Reply:
We need it for dotty website and for backward compatibility, It is also a solution in case of conflicts in one does not want to rename static page or resource.